client/keys/add_ledger_test.go

```diff
@@ -122,6 +122,6 @@ func Test_runAddCmdLedger(t *testing.T) {
 	require.Equal(t, "keyname1", key1.GetName())
 	require.Equal(t, keyring.TypeLedger, key1.GetType())
 	require.Equal(t,
-		"cosmospub1addwnpepqd87l8xhcnrrtzxnkql7k55ph8fr9jarf4hn6udwukfprlalu8lgw0urza0",
-		sdk.MustBech32ifyPubKey(sdk.Bech32PubKeyTypeAccPub, key1.GetPubKey()))
+		"PubKeySecp256k1{034FEF9CD7C4C63588D3B03FEB5281B9D232CBA34D6F3D71AEE59211FFBFE1FE87}",
```

Comment:
I'm curious, any reason why you're using `.String()` here and `legacybech32.MustMarshalPubKey` above?

Reply:
This is a test. We just want to check if the content is good. Look at the 2 strings, which one is easier to reason about? For me the _new_ one because it shows the type name and easy to read hex sequence. Moreover we don't want to use `legacy`  code wherever possible.

Reply:
+1. So let's use `.String()` on line 78 too?

Reply:
I will see, I know that some `String` methods fail.